app/controllers/Application.java

```diff
@@ -1689,6 +1698,168 @@ public static Result restFlowMetricsGraphData(String flowDefId) {
     return ok(new Gson().toJson(sortedDatasets));
   }
 
+  /* Rest Api implementation for providing the tuning parameters details for a job
+   * @param jobId - job execution id passed as a parameter for the GET request
+   * @result JSON containing suggested tuning paramter details
+   * eg: Formatted JSON Data
+    {
+      "tunein":{
+      "id":"https://ltx1-holdemaz01.grid.linkedin.com:8443/executor?execid=1234567&job=dimDataSimpleFlow_dimDataSimple&attempt=0",
+      "jobSuggestedParamSetId":1234,
+      "tuningAlgorithmId":2,
+      "autoApply":false,
+      "tuningAlgorithm":"HBT",
+      "tuningAlgorithms":[
+         {
+            "name":"HBT"
+         },
+         {
+            "name":"OBT"
+         }
+      ],
+      "iterationCount":10,
+      "tuningParameters":[
+         {
+            "name":"mapreduce.map.memory.mb",
+            "jobSuggestedParamValue":"1113.33",
+            "userSuggestedParamValue":"1113.33",
+            "currentParamValue":"1113.33"
+         },
+         {
+            "name":"mapreduce.map.java.opts",
+            "jobSuggestedParamValue":"839",
+            "userSuggestedParamValue":"839",
+            "currentParamValue":"839"
+         }
+      ]
+    }
+   }
+  */
+  public static Result getTuningParameter(String jobId) {
+    logger.info("Getting Tuning Parameters for " + jobId);
```

Comment:
General: This method is a bit long (more than 100 lines) which makes it hard to read. Can you please refactor in smaller methods?

Reply:
Will try to.